codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java

```diff
@@ -372,6 +374,29 @@ public static CharSequence getCharsetAsString(HttpMessage message) {
         return null;
     }
 
+    /**
+     * Fetch MIME type part from message's Content-Type header as an arbitrary char sequence.
+     *
+     * @return the MIME type {@code CharSequence} from message's Content-Type header
+     * or {@code null} if content-type header or MIME type is not presented
+     * <p/>
+     * "content-type: text/html; charset=utf-8" - "text/html" will be returned <br/>
+     * "content-type: text/html" - "text/html" will be returned <br/>
+     * "content-type: " or no header - {@code null} we be returned
+     */
+    public static CharSequence getMimeType(HttpMessage message) {
+        CharSequence contentTypeValue = message.headers().get(HttpHeaderNames.CONTENT_TYPE);
+        if (contentTypeValue != null) {
+            int indexOfSemicolon = AsciiString.indexOfIgnoreCaseAscii(contentTypeValue, SEMICOLON, 0);
```

Comment:
I shouldn't because ByteProcessor need to compare bytes, but we already have chars - no need to make conversion to bytes for all chars in our CharSequence. Solved it another way, made method AsciiString#indexOf(CharSequence, char, startPos) with optimized version for AsciiString CharSequence. But we not will have String here (version of indexOf from String will be actually used). @normanmaurer @Scottmitch Please, revisit PR, it became much bigger.